### ISSUE TITLE
Terminal: No place to open terminal to

### ISSUE THREAD

**Issue by onmyway133** [wbond/sublime_terminal#91]
I installed `Terminal` package, hit Command+Shift+t on a file, and got this error

`Terminal: No place to open terminal to`

I use Sublime Text 2 on Yosemite 10.10.2

**Comment by twolfson** [comment]
I am on vacation for 2 weeks. I don't know any solution for this right now. I will reply in 2 weeks.

**Comment by wbond** [comment]
Was it a text file? What version of sublime text and on what platform?

**Comment by twolfson** [comment]
It's been 1 month of inactivity from @onmyway133 closing issue for now.

**Comment by nathanbell** [comment]
Having this issue too.

**Comment by twolfson** [comment]
@nathanbell Can you give us more steps to reproduce the error?

- What OS are you using?
- What version of Sublime Text are you on?
- How did you receive the error? (e.g. press a shortcut when viewing a file)

**Comment by chen57237861** [comment]
Hi,
I'm hitting the same issue.

  >What OS are you using? Fedora 22
  >What version of Sublime Text are you on? Build 3083
  >How did you receive the error? (e.g. press a shortcut when viewing a file) Input Ctrl+Shift+T

Thanks,
Chen

**Comment by twolfson** [comment]
@chen57237861 Can you provide more information about when the error occurred?

- Did you have any folders in your sidebar?
- Did you have an file buffer open?

**Comment by chen57237861** [comment]
Hi @twolfson 

Thank you for your reply.

It seems that re-installation solves the problem... But there should not be the problem of the package itself but the difference of my environment before and after the re-installation.

Thank you !
Chen

**Comment by twolfson** [comment]
What was it that you reinstalled (e.g. the package, Sublime Text)?

Also, did you open any files during reinstallation?

**Comment by zburns12** [comment]
I just had this same issue. Brand new install of Sublime Text 3 on El Capitan. Uninstalling and reinstalling the Terminal package fixed it. Unfortunately, didn't see this until after I fixed it, but wanted to let you know what exactly was reinstalled.

**Comment by twolfson** [comment]
Cool, thanks for the info. I will try to reproduce by using a fresh machine on Sauce Labs by the end of next weekend.

**Comment by katmoon** [comment]
I have this error when I try to open Terminal on an empty tab. So there is no folder defined where Terminal should start.

**Comment by twolfson** [comment]
@katmoon I believe that was the original edge case the error was for. Where would expect the terminal to open in that scenario? Maybe we could simply invoke the terminal command and let it open in its default directory?

**Comment by katmoon** [comment]
Yes, any default directory would be nice.

I didn't notice the tab was empty in the beginning and thought Terminal didn't work at all, googled the issue, found this one and subscribed. Only after some time noticed that I tried to open it in an empty tab, that's why it didn't work.

**Comment by twolfson** [comment]
@wbond Thoughts on opening in the default `cwd` (probably Sublime's directory) or home directory?

**Comment by katmoon** [comment]
I would prefer home directory. Terminals usually start there.

**Comment by krisdestruction** [comment]
I also think this needs a better error message. Perhaps something like:
`File is not saved. Cannot open a terminal here.`

**Comment by hickford** [comment]
To reproduce:

1. Open Sublime Text
2. File / New File
3. Right click in text editor, 'Open terminal here...'

Expected: Menu item 'Open terminal here' is greyed out (disabled)

What actually happens: You get a silly error "Terminal: No place to open terminal to"

**Comment by twolfson** [comment]
@hickford That would be an enhancement and is unrelated to this issue. Please open a separate issue or a PR.

**Comment by hickford** [comment]
Done https://github.com/wbond/sublime_terminal/issues/119

**Comment by DVN-Anakin** [comment]
Guys, this issue can be fixed by saving your opened file. When You start SublimeText and open some file make sure this file is saved, only after that You can use ctrl+shift+T for opening the Terminal.

**Comment by MamoniAL** [comment]
set defult on settings terminal end download sublime 3

**Comment by maegul** [comment]
On macOS Sierra, ST2 (2221), installed Terminal plug with in past few days.

@DVN-Anakin
I get this error even when the file is saved, but the file/tab is empty.

Adding a single character, without saving, solves issue and opens a terminal.

Quite minor, but I only know this because I wanted to open the terminal before adding some content.  And the error message lead me down the wrong path in wondering what was wrong (I thought I had done something terrible with my git repo).

**Comment by TMorville** [comment]
I get the same error ``Terminal: No place to open terminal to`` , even though the file is saved and has content.

**Comment by twolfson** [comment]
What's the shortcut you're using and is the file in a project folder?

**Comment by TMorville** [comment]
My shortcut is shift-command-T and if you're referring to the file I have open, it's in ``~/tmp/``.

**Comment by twolfson** [comment]
So the code that generates the message is:

https://github.com/wbond/sublime_terminal/blob/1.18.0/Terminal.py#L123-L134

If you're invoking via a shortcut, then there should be no `paths` parameter. This means something is going wrong with `active_view().file_name()` or `folders()`. Can you try opening Sublime Text's console and running those to see what's causing the issue?

**Comment by TMorville** [comment]
OK. For some reason, opening a console allows the shortcut to work... I will see if this is a stable result after some use.

**Comment by gumblur** [comment]
thanks you
